### PR TITLE
Add function to respond to user removal requests

### PR DESCRIPTION
**Pull request by marten**
Can someone sanity-check if there might be more places we need to strip information?

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -318,4 +318,20 @@ def no_other_logins_exist?(login)
...
+    self.email = "noreply-#{SecureRandom.hex(4)}@zooniverse.org"
+    self.current_sign_in_ip = nil
+    self.last_sign_in_ip = nil
+    self.display_name = self.id
+    self.login = self.id
+    self.credited_name = nil
+    self.password = self.password_confirmation = SecureRandom.hex(16)
+    self.global_email_communication = false
+    self.project_email_communication = false
+    self.beta_email_communication = false
```
Comment (astopy) [inline app/models/user.rb]: Could you also set `self.valid_email = false`?

File: app/models/user.rb [hunk 2]
```diff
@@ -318,4 +318,20 @@ def no_other_logins_exist?(login)
   def uploaded_subjects_count
     Subject.unscoped.where(upload_user_id: self.id).count
   end
+
+  def anonymize
+    self.email = "noreply-#{SecureRandom.hex(4)}@zooniverse.org"
+    self.current_sign_in_ip = nil
+    self.last_sign_in_ip = nil
+    self.display_name = self.id
+    self.login = self.id
```
Comment (astopy) [inline app/models/user.rb]: I know it's unlikely, but what if there's already a user with this login?

File: app/models/user.rb (hunk 1 above)
Comment (astopy) [inline app/models/user.rb]: Maybe also set `self.private_profile = true`

File: app/models/user.rb [hunk 3]
```diff
@@ -318,4 +318,20 @@ def no_other_logins_exist?(login)
   def uploaded_subjects_count
     Subject.unscoped.where(upload_user_id: self.id).count
   end
+
+  def anonymize
+    self.email = "noreply-#{SecureRandom.hex(4)}@zooniverse.org"
+    self.current_sign_in_ip = nil
+    self.last_sign_in_ip = nil
+    self.display_name = self.id
```
Comment (astopy) [inline app/models/user.rb]: This might be visible publicly in some places, so I wonder if something like `"Deleted user #{self.id}"` would be better.

**Comment by camallen** [comment]
https://github.com/zooniverse/Panoptes/blob/master/lib/user_info_scrubber.rb

The original plan was to use the above and write it up to the users#destroy method. We should probably merge these two bits of code

**Comment by camallen** [comment]
Talk comments should stay, we should just show the user name as deleted or the like

File: app/models/user.rb (hunk 2 above)
Comment (marten) [inline app/models/user.rb]: Validation and indexes would stop us from saving that. Not too worried about that, unlikely to happen and it's a behind the scenes management function anyways.

**Pull request merged**